hnn_core/gui/gui.py

```diff
@@ -663,6 +694,10 @@ def compose(self, return_layout=True):
                 self.widget_ntrials, self.widget_backend_selection,
                 self._backend_config_out]),
         ], layout=self.layout['config_box'])
+        # Displays the default backend options
+        handle_backend_change(self.widget_backend_selection.value,
+                              self._backend_config_out, self.widget_mpi_cmd,
+                              self.widget_n_jobs)
```

Comment:
I was actually suggesting something simpler ... hnn-core already defaults to JoblibBackend if MPI is not available, so all you had to do was expose MPIBackend and get rid of the JoblibBackend. Anyhow @dylansdaniels do you want this merged for the GUI tutorial today?

Reply:
That handler in the gui changes the displayed options based on the backend drop-down it doesn't switch between backends. In hnn-core it MPIBackend only switches to joblib is it's a single process, I don't think it checks for MPI and reverts to joblib if it can't find it.

Reply:
Nevermind, you're right. It does revert to joblib. Though it runs on joblib with only 1 processor, is there a reason why it only uses 1?

Reply:
I think it's because specifying multiple processes with the joblib backend, unlike the MPI backend, deosn't make sense for all types of simulations. Since joblib parallelizes over multiple trials, it only works for simulations that include multiple trials. Plus parallel joblib processes are memory intensive, so silently switching from MPI to joblib can produce errors if the user wasn't anticipating needing the extra resources needed for a joblib task with many trials.

Reply:
I agree with @rythorpe . I would say the GUI should just use MPIBackend, and if not available, just run everything in a single core. It's probably a misnomer to call it "JoblibBackend" if it's running on a single core. Hence, I wouldn't bother exposing it to user.

Reply:
Do we really want to strip the functionally of using multicores with joblib? It's a nice feature for those who don't have MPI implemented. On my machine a 3 trial simulation ran in less than half the time on 3 cores than 1 core.

Reply:
The point is I have never seen anyone run multiple trials in the GUI ... but I will defer to the judgement of @rythorpe and @ntolley as they are the real users

Reply:
I'm on the edge honestly. 9/10 times GUI users will be using `MPIBackend`, which applies to all types of simulations provided that it's installed correctly. IDK that providing granular control of `JoblibBackend` is worth it, but I could be convinced otherwise.

Reply:
Do we know if anyone who is actually using `JoblibBackend` in the GUI for the their research? I personally don't know of anyone and am generally under the impression that folks who prefer the GUI would want a general-use parallel backend that works without too much fuss.

Your work @ntolley is unique in that your questions require lots of parameter sweeps, which are best served by using the API directly. Note that limited RAM is only an issue for Joblib, which I'm arguing should essentially default to running a single job within the context of the GUI.

Reply:
Pretty sure a lot of Darcy's initial work compared over multiple trials, although I'll concede she's using the API now. If you're doing 1-1 comparisons of ERP's for instance it's definitely more robust to compute an average over multiple trials. In tutorials we use 1 trial because it's efficient, but that isn't necessarily how you should fit these signals to data.

Reply:
Again I totally agree with making MPI with oversubscription the default, but I genuinely think trial-based parallelism is a useful feature even in the GUI setting! 

These are my cards on the table :smile:, but if I get outvoted I'll concede

Reply:
Fair enough. If GUI users use it, we should support it.

Reply:
I personally feel that once a feature is added, users find ways to justify keeping it. That's why one should start with minimal features first and then add based on necessity and feedback from user base. Anyway, call is yours @ntolley !